modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js

```diff
@@ -615,8 +615,10 @@ class Iframe extends React.Component {
 	}
 
 	componentWillUnmount() {
-		if (this.beforeScreenFlipHandler) {
-			Liferay.detach(this.beforeScreenFlipHandler);
+		if (this.spaNavigationHandlers) {
```

Comment:
Should we check for length too? OR does that even matter since this is on unmount, which theoretically only happens once.

Reply:
There is probably no need. There is no case where length could be zero, it's either two or array is `undefined`.